cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/TimePartitionedFileSetDefinition.java

```diff
@@ -82,7 +86,16 @@ public PartitionedFileSet getDataset(DatasetContext datasetContext, DatasetSpeci
     if (time != null) {
       // set the output path according to partition time
       if (FileSetArguments.getOutputPath(arguments) == null) {
-        String path = String.format("%tF/%tH-%tM.%d", time, time, time, time);
+        String outputPathFormat = TimePartitionedFileSetArguments.getOutputPathFormat(arguments);
+        String path;
+        if (Strings.isNullOrEmpty(outputPathFormat)) {
+          path = String.format("%tF/%tH-%tM.%d", time, time, time, time);
+        } else {
+          SimpleDateFormat format = new SimpleDateFormat(outputPathFormat);
+          String timeZoneID = TimePartitionedFileSetArguments.getOutputPathTimeZone(arguments);
+          format.setTimeZone(TimeZone.getTimeZone("" + timeZoneID));
```

Comment:
o in case timeZoneID is null?  it's more clear if you explicitly handle that case. Otherwise somebody who is not familiar with the getTimeZone will get confused.